vdev/src/commands/build.rs

```diff
@@ -0,0 +1,56 @@
+use clap::Args;
+
+use crate::app::Application;
+
+/// Build Vector
+#[derive(Args, Debug)]
+#[command()]
+pub struct Cli {
+    /// The build target e.g. x86_64-unknown-linux-musl
+    target: Option<String>,
+
+    /// Build with optimizations
+    #[arg(short, long)]
+    release: bool,
+
+    /// The feature to activate (multiple allowed)
+    #[arg(short = 'F', long)]
+    feature: Vec<String>,
+}
+
+impl Cli {
+    pub fn exec(&self, app: &Application) {
+        let mut command = app.command("cargo");
+        command.args(["build", "--no-default-features"]);
+
+        if self.release {
+            command.arg("--release");
+        }
+
+        command.arg("--features");
+        if !self.feature.is_empty() {
+            command.args([self.feature.join(",")]);
+        } else {
+            if app.platform.windows() {
+                command.arg("default-msvc");
+            } else {
+                command.arg("default");
+            }
+        };
+
+        if let Some(target) = self.target.as_deref() {
+            command.args(["--target", target]);
+        } else {
+            if app.platform.windows() {
```

Comment:
We'd also need handling for Apple?

Reply:
Yup! Was just copying https://github.com/vectordotdev/vector/blob/547439889021192aacb11e71de2b3a02f81edde5/Makefile#L7-L18

Reply:
Cool, just making sure I didn't miss some magic around cross compiling 😆

Reply:
also btw `x86_64-unknown-windows-msvc` doesn't exist so I fixed that 😅

Reply:
It's _really_ unknown.